docs/index.rst

```diff
@@ -47,6 +47,7 @@ Contents:
     references
     creating
     faq
+    types
```

Comment:
Any reason to add this as an entirely new document? If not might just be reasonable to put this in the `validate.rst` doc, I think even format is already just in that same doc.